src/pgduckdb_types.cpp

```diff
@@ -495,6 +519,53 @@ ChildTypeFromArray(Oid array_type) {
 	}
 }
 
+duckdb::LogicalType
+ConvertPostgresEnumToDuckEnum(Oid enum_type_oid) {
+	/* Get the list of existing members of the enum */
+	auto list =
+	    PostgresFunctionGuard<CatCList *>([](int cacheId, Datum key) { return SearchSysCacheList1(cacheId, key); },
+	                                      ENUMTYPOIDNAME, ObjectIdGetDatum(enum_type_oid));
+	auto nelems = list->n_members;
+
+	/* Sort the existing members by enumsortorder */
+	std::vector<HeapTuple> enum_members(nelems);
+	for (int i = 0; i < nelems; ++i) {
+		enum_members[i] = &list->members[i]->tuple;
+	}
+
+	auto sort_order_cmp = [](const HeapTuple &v1, const HeapTuple &v2) {
+		Form_pg_enum en1 = (Form_pg_enum)GETSTRUCT(v1);
+		Form_pg_enum en2 = (Form_pg_enum)GETSTRUCT(v2);
+
+		if (en1->enumsortorder < en2->enumsortorder) {
+			return true; // v1 < v2
+		} else if (en1->enumsortorder > en2->enumsortorder) {
+			return false; // v1 > v2
+		} else {
+			return false; // v1 == v2
+		}
+	};
+
+	std::sort(enum_members.begin(), enum_members.end(), sort_order_cmp);
+
+	idx_t allocation_size = enum_members.size();
+	allocation_size += enum_members.size() / 4;
+	allocation_size += (enum_members.size() % 4) != 0;
```

Comment:
This definitely needs a code comment to explain what's going on. Afaict you are allocating enough additional space some extra space after. And I'm guessing the 4 is for the size of the `Oid`? If so let's make that clear:
```suggestion
	allocation_size += enum_members.size() / sizeof(Oid);
	allocation_size += (enum_members.size() % sizeof(Oid)) != 0;
```

Does this take care of allignment?

Reply:
I'm not sure, but `string_t` is 16 bytes so I assume the answer is no since it's divisable by 8 (`sizeof(void *)`) ?